src/main/java/org/generationcp/middleware/service/impl/user/UserServiceImpl.java

```diff
@@ -156,15 +151,36 @@ public Integer createUser(final UserDto userDto) {
 		user.setType(0);
 
 		// Add user roles to the particular user
-		user.setRoles(Arrays.asList(new UserRole(user, userDto.getRole())));
+		final List<UserRole> userRoles = new ArrayList<>();
+		if (userDto.getUserRoles() != null) {
+			for (final UserRoleDto userRoleDto : userDto.getUserRoles()) {
+				final UserRole userRole = new UserRole();
+				final Role role = new Role();
+				role.setId(userRoleDto.getRole().getId());
+				userRole.setRole(role);
+				userRole.setUser(user);
+				if (userRoleDto.getCrop() != null) {
+					userRole.setCropType(new CropType(userRoleDto.getCrop().getCropName()));
+				}
+				if (userRoleDto.getProgram() != null) {
+					final Project project =
+						this.workbenchDaoFactory.getProjectDAO().getByUuid(userRoleDto.getProgram().getUuid(), userRoleDto.getCrop().getCropName());
```

Comment:
Maybe we can build a map of project UUIDs and do one-off query of Projects by UUID instead of doing it inside the loop